src/main/java/mate/academy/spring/validation/EmailValidator.java

```diff
@@ -0,0 +1,21 @@
+package mate.academy.spring.validation;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+public class EmailValidator implements ConstraintValidator<Email, String> {
+    private static final String EMAIL_PATTERN = "^[\\w!#$%&’*+/=?`{|}~^-]+(?:\\.[\\w!#$%&’*+/=?"
+             + "`{|}~^-]+)*@(?:[a-zA-Z\\d-]+\\.)+[a-zA-Z]{2,6}$";
+
+    @Override
+    public boolean isValid(String email, ConstraintValidatorContext constraintValidatorContext) {
+        if (email.isEmpty()) {
+            return false;
+        }
+        Pattern pattern = Pattern.compile(EMAIL_PATTERN);
+        Matcher matcher = pattern.matcher(email);
+        return matcher.matches();
```

Comment:
```suggestion
        email != null && Pattern.compile(EMAIL_PATTERN).matcher(email).matches();
```

we could make it simpler